dipy/reconst/dki.py

```diff
@@ -2336,6 +2349,255 @@ def wls_fit_dki(design_matrix, data):
     return dki_params
 
 
+def _cls_iter(sdp, design_matrix, sig, cvxpy_solver=None):
+    """ Helper function used by cls_fit_dki - Applies an LMI constrained
+    OLS fit of the diffusion kurtosis model to single voxel signals.
+
+    Parameters
+    ----------
+    sdp : PositiveDefiniteLeastSquares instance
+        A CVXPY representation of a regularized least squares optimization
+        problem.
+    design_matrix : array (g, 22)
+        Design matrix holding the covariants used to solve for the regression
+        coefficients.
+    sig : array (g,)
+        Diffusion-weighted signal for a single voxel data.
+    cvxpy_solver : str, optional
+        cvxpy solver name. Optionally optimize the positivity constraint with a
+        particular cvxpy solver. See http://www.cvxpy.org/ for details.
+        Default: None (cvxpy chooses its own solver).
+
+    Returns
+    -------
+    dki_params : array (27,)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follows:
+            1) Three diffusion tensor's eigenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+
+    """
+
+    # DKI ordinary linear least square solution
+    log_s = np.log(sig)
+    result = sdp.solve(design_matrix, log_s, check=True, solver=cvxpy_solver)
+
+    # Extracting the diffusion tensor parameters from solution
+    DT_elements = result[:6]
+    evals, evecs = decompose_tensor(from_lower_triangular(DT_elements))
+
+    # Extracting kurtosis tensor parameters from solution
+    MD_square = (evals.mean(0))**2
+    if MD_square:
+        KT_elements = result[6:21] / MD_square
+    else:
+        KT_elements = 0.*result[6:21]
```

Comment:
To be more pythonic, Can you use a one-liner `KT_elements = result[6:21] / MD_square if MD_square else 0.*result[6:21]`